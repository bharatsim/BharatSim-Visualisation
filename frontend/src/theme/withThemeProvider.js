import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme(props) {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent {...props} />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
