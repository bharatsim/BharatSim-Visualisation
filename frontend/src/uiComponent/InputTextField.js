import TextField from '@material-ui/core/TextField';
import React from 'react';
import { Box, Grid } from '@material-ui/core';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => {
  return {
    helperText: {
      margin: theme.spacing(1, 0, 0, 0),
    },
    inputField: {
      padding: theme.spacing(4),
      backgroundColor: theme.colors.grayScale['100'],
      borderRadius: theme.spacing(1),
    },
    textFieldContainer: {
      width: '100%',
      display: 'flex',
      justifyContent: 'space-between',
    },
    labelContainer: {
      display: 'flex',
      paddingTop: theme.spacing(4),
      justifyContent: 'flex-end',
      height: '100%',
    },
    label: {
      wordWrap: 'break-word',
    },
  };
});
function InputTextField({ label, id, value, helperText, error, onChange }) {
  const classes = useStyles();
  return (
    <Grid container xl={12}>
      <Grid item xs={3}>
        <Box className={classes.labelContainer}>
          <label htmlFor={id} className={classes.label}>
            {label}
          </label>
        </Box>
      </Grid>
      <Grid item xs={1} />
      <Grid item xs={7}>
        <TextField
          id={id}
          value={value}
          helperText={error || helperText}
          error={!!error}
          variant="filled"
          fullWidth
          InputProps={{ classes: { input: classes.inputField } }}
          FormHelperTextProps={{ classes: { root: classes.helperText } }}
          onChange={onChange}
        />
      </Grid>
    </Grid>
  );
}

InputTextField.propTypes = {
  label: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  helperText: PropTypes.string,
  error: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

InputTextField.defaultProps = {
  helperText: '',
  error: '',
};

export default InputTextField;
