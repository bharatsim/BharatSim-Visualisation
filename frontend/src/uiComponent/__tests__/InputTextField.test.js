import { render } from '@testing-library/react';
import React from 'react';
import { fireEvent } from '@testing-library/dom';
import InputTextField from '../InputTextField';
import withThemeProvider from '../../theme/withThemeProvider';

describe('InputTextField', () => {
  const InputTextFieldComponent = withThemeProvider(InputTextField);
  it('should render input text field with for given information', () => {
    const { container } = render(
      <InputTextFieldComponent
        onChange={jest.fn()}
        id="input-id"
        label="input-label"
        value="input-value"
        error=""
        helperText="helper-text"
      />,
    );
    expect(container).toMatchSnapshot();
  });
  it('should change the value on change for input field', () => {
    const mockOnChange = jest.fn();
    const { getByLabelText } = render(
      <InputTextFieldComponent
        onChange={mockOnChange}
        id="input-id"
        label="input-label"
        value="input-value"
        error=""
        helperText="helper-text"
      />,
    );
    fireEvent.change(getByLabelText('input-label'), {
      target: { value: 'new input value' },
    });
    expect(mockOnChange).toHaveBeenCalled();
  });
  it('should show error if any errors are passed in prop ', () => {
    const { getByText } = render(
      <InputTextFieldComponent
        onChange={jest.fn()}
        id="input-id"
        label="input-label"
        value="input-value"
        error="Some error has been created"
        helperText="helper-text"
      />,
    );
    expect(getByText('Some error has been created')).not.toBe(null);
  });
});
