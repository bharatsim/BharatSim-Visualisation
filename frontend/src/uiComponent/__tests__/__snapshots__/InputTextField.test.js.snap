// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputTextField should render input text field with for given information 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-grid-xl-12"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      <div
        class="MuiBox-root MuiBox-root-6 makeStyles-labelContainer-4"
      >
        <label
          class="makeStyles-label-5"
          for="input-id"
        >
          input-label
        </label>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
    />
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <div
          class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-describedby="input-id-helper-text"
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input makeStyles-inputField-2"
            id="input-id"
            type="text"
            value="input-value"
          />
        </div>
        <p
          class="MuiFormHelperText-root makeStyles-helperText-1 MuiFormHelperText-contained MuiFormHelperText-filled"
          id="input-id-helper-text"
        >
          helper-text
        </p>
      </div>
    </div>
  </div>
</div>
`;
